tree/dataframe/src/RSqliteDS.cxx

```diff
@@ -13,19 +13,304 @@
+{
+   return SQLITE_OPEN_READONLY;
+}
+
+static int VfsRdOnlyTruncate(sqlite3_file * /*pFile*/, sqlite_int64 /*size*/)
+{
+   return SQLITE_OPEN_READONLY;
+}
+
+static int VfsRdOnlySync(sqlite3_file * /*pFile*/, int /*flags*/)
+{
+   return SQLITE_OK;
+}
+
+////////////////////////////////////////////////////////////////////////////
+/// Returns the cached file size
+static int VfsRdOnlyFileSize(sqlite3_file *pFile, sqlite_int64 *pSize)
+{
+   VfsRootFile *p = reinterpret_cast<VfsRootFile *>(pFile);
+   *pSize = p->size;
+   return SQLITE_OK;
+}
+
+static int VfsRdOnlyLock(sqlite3_file * /*pFile*/, int /*level*/)
+{
+   return SQLITE_OK;
+}
+
+static int VfsRdOnlyUnlock(sqlite3_file * /*pFile*/, int /*level*/)
+{
+   return SQLITE_OK;
+}
+
+static int VfsRdOnlyCheckReservedLock(sqlite3_file * /*pFile*/, int *pResOut)
+{
+   *pResOut = 0;
+   return SQLITE_OK;
+}
+
+static int VfsRdOnlyFileControl(sqlite3_file * /*p*/, int /*op*/, void * /*pArg*/)
+{
+   return SQLITE_NOTFOUND;
+}
+
+static int VfsRdOnlySectorSize(sqlite3_file * /*pFile*/)
+{
+   return SQLITE_OPEN_READONLY;
+}
+
+static int VfsRdOnlyDeviceCharacteristics(sqlite3_file * /*pFile*/)
+{
+   return SQLITE_OPEN_READONLY;
+}
+
+////////////////////////////////////////////////////////////////////////////
+/// Fills a new VfsRootFile struct enclosing a Davix file
+static int VfsRdOnlyOpen(sqlite3_vfs * /*vfs*/, const char *zName, sqlite3_file *pFile, int flags, int * /*pOutFlags*/)
+{
+   VfsRootFile *p = new (pFile) VfsRootFile();
+   p->pFile.pMethods = nullptr;
+
+   static const sqlite3_io_methods io_methods = {
+      1, // version
+      VfsRdOnlyClose,
+      VfsRdOnlyRead,
+      VfsRdOnlyWrite,
+      VfsRdOnlyTruncate,
+      VfsRdOnlySync,
+      VfsRdOnlyFileSize,
+      VfsRdOnlyLock,
+      VfsRdOnlyUnlock,
+      VfsRdOnlyCheckReservedLock,
+      VfsRdOnlyFileControl,
+      VfsRdOnlySectorSize,
+      VfsRdOnlyDeviceCharacteristics,
+      // Version 2 and later callbacks
+      nullptr, // xShmMap
+      nullptr, // xShmLock
+      nullptr, // xShmBarrier
+      nullptr, // xShmUnmap
+      nullptr, // xFetch
+      nullptr  // xUnfetch
+   };
+
+   if (flags & SQLITE_OPEN_READWRITE)
+      return SQLITE_IOERR;
+   if (flags & SQLITE_OPEN_DELETEONCLOSE)
+      return SQLITE_IOERR;
+   if (flags & SQLITE_OPEN_EXCLUSIVE)
+      return SQLITE_IOERR;
+
+   Davix::DavixError *err = NULL;
+   p->fd = p->pos.open(NULL, zName, O_RDONLY, &err);
+
+   if (!p->fd) {
+      ::Error("VfsRdOnlyOpen", "%s\n", err->getErrMsg().c_str());
+      return SQLITE_IOERR;
+   }
+
+   struct stat buf;
+   if (p->pos.stat(NULL, zName, &buf, NULL) == -1) {
+      return SQLITE_IOERR;
+   }
+   p->size = buf.st_size;
+
+   p->pFile.pMethods = &io_methods;
+   return SQLITE_OK;
+}
+
+static int VfsRdOnlyDelete(sqlite3_vfs *__attribute__((unused)), const char * /*zName*/, int /*syncDir*/)
```

Comment:
why does _this particular_ argument require a `__attribute__((unused))` (but not all others)?